libmamba/src/solv-cpp/queue.cpp

```diff
@@ -0,0 +1,288 @@
+        : ObjQueue(other.begin(), other.end())
+    {
+    }
+
+    ObjQueue::~ObjQueue()
+    {
+        if (data() != nullptr)
+        {
+            queue_free(get());
+        }
+    }
+
+    auto ObjQueue::operator=(ObjQueue&& other) -> ObjQueue&
+    {
+        swap(*this, other);
+        // Leaving other empty to make sure ressources are no longer used
+        auto empty = ObjQueue(nullptr);
+        swap(other, empty);
+        return *this;
+    }
+
+    auto ObjQueue::operator=(ObjQueue const& other) -> ObjQueue&
+    {
+        auto other_copy = ObjQueue(other);
+        swap(*this, other_copy);
+        return *this;
+    }
+
+    void ObjQueue::push_back(value_type id)
+    {
+        queue_push(get(), id);
+    }
+
+    void ObjQueue::push_back(value_type id1, value_type id2)
+    {
+        queue_push2(get(), id1, id2);
+    }
+
+    auto ObjQueue::insert(const_iterator pos, value_type id) -> iterator
+    {
+        auto const offset = offset_of(pos);
+        insert(offset, id);
+        return begin() + offset;
+    }
+
+    auto ObjQueue::capacity() const -> size_type
+    {
+        return static_cast<size_type>(m_queue.count + m_queue.left);
+    }
+
+    auto ObjQueue::size() const -> size_type
+    {
+        assert(m_queue.count >= 0);
+        return static_cast<std::size_t>(m_queue.count);
+    }
+
+    auto ObjQueue::empty() const -> bool
+    {
+        return size() == 0;
+    }
+
+    auto ObjQueue::erase(const_iterator pos) -> iterator
+    {
+        auto const offset = offset_of(pos);
+        assert(offset <= std::numeric_limits<int>::max());
+        queue_delete(get(), static_cast<int>(offset));
+        return begin() + offset;
+    }
+
+    void ObjQueue::reserve(size_type new_cap)
+    {
+        if (new_cap < capacity())
+        {
+            return;
+        }
+        size_type cap_diff = new_cap - capacity();
+        assert(cap_diff <= std::numeric_limits<int>::max());
+        queue_prealloc(get(), static_cast<int>(cap_diff));
+    }
+
+    void ObjQueue::clear()
+    {
+        queue_empty(get());
+    }
+
+    auto ObjQueue::front() -> reference
+    {
+        return *begin();
+    }
+
+    auto ObjQueue::front() const -> const_reference
+    {
+        return *begin();
+    }
+
+    auto ObjQueue::back() -> reference
+    {
+        return *rbegin();
+    }
+
+    auto ObjQueue::back() const -> const_reference
+    {
+        return *rbegin();
+    }
+
+    auto ObjQueue::operator[](size_type pos) -> reference
+    {
+        return data()[pos];
+    }
+
+    auto ObjQueue::operator[](size_type pos) const -> const_reference
+    {
+        return data()[pos];
+    }
+
+    namespace
+    {
+        void ensure_valid_pos(ObjQueue::size_type size, ObjQueue::size_type pos)
+        {
+            if (pos >= size)
+            {
+                std::stringstream ss = {};
+                ss << "Index " << pos << " is greater that the number of elements (" << size << ')';
```

Comment:
Use fmt here (maybe with a comment to use `std::format` once C++20 is available)

Reply:
The idea is to keep these wrappers dependency free so I think we should avoid using `fmt` until there is some strong need for it.